typed-racket-lib/typed-racket/private/type-contract.rkt

```diff
@@ -399,8 +407,13 @@
         (if (from-typed? typed-side)
             (and/sc sc any-wrap/sc)
             sc))
+      ;(eprintf "predef: ~s ~s ~s\n" predef-contracts type typed-side)
       (cached-match
        sc-cache type typed-side
+       [(app (lambda (t) (hash-ref predef-contracts (cons t typed-side) #f))
+             (? values con-id))
+        ;(eprintf "found a match ~s ~s\n" con-id type)
+        (impersonator/sc (syntax-local-introduce con-id))]
```

Comment:
Instead of always making an `impersonator/sc`, will this eventually check the kind and make a flat or chaperone when possible?

Reply:
Yes.